--- zopfli-zopfli-1.0.3/Makefile.orig	2023-07-01 18:35:05.780549484 +0200
+++ zopfli-zopfli-1.0.3/Makefile	2023-07-01 18:38:20.396161828 +0200
@@ -1,7 +1,7 @@
 CC ?= gcc
 CXX ?= g++
 
-override CFLAGS := -W -Wall -Wextra -ansi -pedantic -lm -O2 -Wno-unused-function -fPIC $(CFLAGS)
+override CFLAGS := -W -Wall -Wextra -ansi -pedantic -O2 -Wno-unused-function -fPIC $(CFLAGS)
 override CXXFLAGS := -W -Wall -Wextra -ansi -pedantic -O2 -fPIC $(CXXFLAGS)
 
 ZOPFLILIB_SRC = src/zopfli/blocksplitter.c src/zopfli/cache.c\
@@ -38,11 +38,11 @@ obj/%.o: %.cpp
 
 # Zopfli binary
 zopfli: $(ZOPFLILIB_OBJ) $(ZOPFLIBIN_OBJ)
-	$(CC) $^ $(CFLAGS) -o $@ $(LDFLAGS)
+	$(CC) $(LDFLAGS) $^ $(CFLAGS) -o $@ -lm
 
 # Zopfli shared library
 libzopfli: $(ZOPFLILIB_OBJ)
-	$(CC) $^ $(CFLAGS) -shared -Wl,-soname,libzopfli.so.1 -o libzopfli.so.1.0.3 $(LDFLAGS)
+	$(CC) $(LDFLAGS) $^ $(CFLAGS) -shared -Wl,-soname,libzopfli.so.1 -o libzopfli.so.1.0.3 -lm
 
 # Zopfli static library
 libzopfli.a: $(ZOPFLILIB_OBJ)
@@ -50,11 +50,11 @@ libzopfli.a: $(ZOPFLILIB_OBJ)
 
 # ZopfliPNG binary
 zopflipng: $(ZOPFLILIB_OBJ) $(LODEPNG_OBJ) $(ZOPFLIPNGLIB_OBJ) $(ZOPFLIPNGBIN_OBJ)
-	$(CXX) $^ $(CFLAGS) -o $@ $(LDFLAGS)
+	$(CXX) $(LDFLAGS) $^ $(CFLAGS) -o $@
 
 # ZopfliPNG shared library
 libzopflipng: $(ZOPFLILIB_OBJ) $(LODEPNG_OBJ) $(ZOPFLIPNGLIB_OBJ)
-	$(CXX) $^ $(CFLAGS) --shared -Wl,-soname,libzopflipng.so.1 -o libzopflipng.so.1.0.3 $(LDFLAGS)
+	$(CXX) $(LDFLAGS) $^ $(CFLAGS) --shared -Wl,-soname,libzopflipng.so.1 -o libzopflipng.so.1.0.3
 
 # ZopfliPNG static library
 libzopflipng.a: $(LODEPNG_OBJ) $(ZOPFLIPNGLIB_OBJ)
